core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesStore.scala

```diff
@@ -73,8 +74,12 @@ extends TimeSeriesStore with StrictLogging {
     }
   }
 
-  def scanTsCardinalities(ref: DatasetRef, shards: Seq[Int],
+  def scanTsCardinalities(queryContext: QueryContext, ref: DatasetRef, shards: Seq[Int],
                           shardKeyPrefix: Seq[String], depth: Int): Seq[CardinalityRecord] = {
+    // adding an additional check to verify if the partition is same as mentioned in query context
+    require(isCorrectPartitionForCardinalityQuery(queryContext, FILODB_PARTITION),
```

Comment:
It seems `queryContext` is only used for this newly added check.
Should this check happen before this function?

Reply:
Yes. As of now, query context is not being used in TsCardinalities.  Thought of multiple ways, but querycontext was easier to pass without impacting the existing code flow.

We need this check when it is being executed on the filodb ( and also the component should have access to the config ). Before this, we are still in logical / exec plan realm.